ext/standard/file.c

```diff
@@ -1467,6 +1467,62 @@ PHP_FUNCTION(unlink)
 }
 /* }}} */
 
+/* {{{ Sync file to storage. Similar to fflush() but blocks until OS buffers have flushed. */
+PHP_FUNCTION(fsync)
+{
+	zval *res;
+	int ret;
+	php_stream *stream;
+
+	ZEND_PARSE_PARAMETERS_START(1, 1)
+		Z_PARAM_RESOURCE(res)
+	ZEND_PARSE_PARAMETERS_END();
+
+	PHP_STREAM_TO_ZVAL(stream, res);
+
+	if (!php_stream_sync_supported(stream)) {
+		php_error_docref(NULL, E_WARNING, "Can't fsync this stream!");
+		RETURN_FALSE;
+	}
+
+	ret = php_stream_sync(stream, 0);
+	if (ret) {
+		RETURN_FALSE;
+	}
+	RETURN_TRUE;
```

Comment:
```suggestion
	RETURN_BOOL(php_stream_sync(stream, /* data_only */ 0) == 0);
```
is how I would write this personally.